### PR TITLE
Roster sshknownhosts

### PR DESCRIPTION
**Pull request by remijouannet**
### What does this PR do?

Hello,

I developed a new roster to parse a known_hosts file and generate a roster from it,
i just want to know if that something you would merge, i haven't write the documentation or execute pep8 on the code yet

### Tests written?

not yet

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by remijouannet** [comment]
anyone ?

**Comment by waynew** [comment]
@remijouannet Thanks for the PR! I'll take a look at this some time today!

**Comment by waynew** [comment]
Personally I think the addition is a good idea, and seems like it's shaping up well. Obviously some docs, and some automated tests still need to happen, but otherwise seems like a useful addition.

@saltstack/team-core Should salt-ssh be able to build the roster from `~/.ssh/known_hosts`? I'm 👍 on the idea. Agree? Disagree?

**Comment by remijouannet** [comment]
Hi everyone,
just upping this PR since i havn't got any answer

**Comment by waynew** [comment]
Well, I guess there aren't any objections :) Couple of failing checks that may or may not be related.

File: doc/ref/roster/all/salt.roster.sshknownhosts.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+=====================
```
Comment (waynew) [inline doc/ref/roster/all/salt.roster.sshknownhosts.rst]:
```suggestion
=========================
```

File: doc/ref/roster/all/salt.roster.sshknownhosts.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+salt.roster.sshknownhosts
+=====================
```
Comment (waynew) [inline doc/ref/roster/all/salt.roster.sshknownhosts.rst]:
```suggestion
=========================
```

**Review by waynew — CHANGES_REQUESTED** [review]
If I understand the complaint in Jenkins correctly, the surrounding lines should be the same length as their contents.

Give that a shot and see if it sorts the docs issue!

**Review by waynew — CHANGES_REQUESTED** [review]
Hi @remijouannet this is looking great!

The only thing it currently lacks is some tests. You could write unit or integration tests (or both). I *think* that tests/unit/ssh/test_roster_defaults.sh might have a reasonable example test you could look at.

Alternatively, there are also  tests in `tests/integration/ssh/` that you could probably look at 👍 

Once you've got some tests, I think this will be totally ready to go!

**Comment by remijouannet** [comment]
Hi, yes really sorry i was really busy at works, I will work on this as soon as i can

**Comment by remijouannet** [comment]
hi @waynew, one month later, i finally found the time to finish this PR, i did some refacto by using the new utils module "roster_matcher", include a unit test and write the documentation
waiting for your review now

File: salt/roster/sshknownhosts.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+'''
+Parses roster entries out of Host directives from SSH known_hosts
+
+Example of configuration,
+keep in mind that known_hosts file only contain hostname/IP,
+so you have to use roster_defaults for other parameter for each minion
+
+.. code-block:: ini
```
Comment (waynew) [inline salt/roster/sshknownhosts.py]:
```suggestion
.. code-block:: yaml
```

File: salt/roster/sshknownhosts.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+Now you can use the module
+
+.. code-block:: bash
+
+    salt-ssh --roster sshknownhosts '*' -r "echo hi"
+
+Or with a Saltfile
+
+.. code-block:: ini
```
Comment (waynew) [inline salt/roster/sshknownhosts.py]:
```suggestion
.. code-block:: yaml
```

File: salt/roster/sshknownhosts.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+'''
+Parses roster entries out of Host directives from SSH known_hosts
+
+Example of configuration,
```
Comment (waynew) [inline salt/roster/sshknownhosts.py]:
```suggestion
Sample configuration:
```

File: salt/roster/sshknownhosts.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+keep in mind that known_hosts file only contain hostname/IP,
+so you have to use roster_defaults for other parameter for each minion
```
Comment (waynew) [inline salt/roster/sshknownhosts.py]: The previous two lines might read better like this:
```
.. note::

    The ``known_hosts`` file only contains hostname/IP. To pass other parameters, 
    use ``roster_defaults``.

```

**Review by waynew — CHANGES_REQUESTED** [review]
Looks great! Just a couple of doc tweaks and then I'll ✔️ this!

File: salt/roster/sshknownhosts.py (hunk 6 above)
Comment (waynew) [reply salt/roster/sshknownhosts.py]: @remijouannet were you also going to address this change?

Comment (remijouannet) [reply salt/roster/sshknownhosts.py]: Yeah I'm on my phone right now, so I'm gonna wait tomorrow when I have access to a computer

Comment (waynew) [reply salt/roster/sshknownhosts.py]: Great! Looking forward to getting this merged! 🙂

**Review by waynew — APPROVED** [review]

**Comment by waynew** [comment]
🎉

**Pull request merged**